lib/vagrant-libvirt/action/set_boot_order.rb

```diff
@@ -36,21 +36,25 @@ def call(env)
             # Parse the XML and find each defined drive and network interfacee
             hd = xml.search("/domain/devices/disk[@device='disk']")
             cdrom = xml.search("/domain/devices/disk[@device='cdrom']")
-            network = xml.search("/domain/devices/interface[@type='network']")
-
+            network = xml.search("/domain/devices/interface[@type='udp']")
+            network += xml.search("/domain/devices/interface[@type='network']")
```

Comment:
On first glance would this not result in all udp devices being ordered before other network devices instead of the order defined?

I think you can use an xpath predicate of the following to get them in one go and retain the same order they were defined in:

```

            network = xml.search("/domain/devices/interface[@type='network' and @type='udp']")
```